index.js

```diff
@@ -2683,6 +2694,41 @@
         return filter(is(type), Just(x));
       });
 
+  //# keys :: StrMap a -> Array String
+  //.
+  //. Returns an array of the keys of the supplied StrMap.
+  //.
+  //. ```javascript
+  //. > S.keys({a: 1, b: 2, c: 3})
+  //. ['a', 'b', 'c']
+  //. ```
+  S.keys =
+  def('keys',
+      {},
+      [$.StrMap(a), $.Array($.String)],
+      keys);
+
+  //# values :: StrMap a -> Array a
+  //.
+  //. Returns an array of the values of the supplied StrMap.
+  //.
+  //. ```javascript
+  //. > S.values({a: 1, b: 2, c: 3})
+  //. [1, 2, 3]
+  //. ```
+  S.values =
+  def('values',
+      {},
+      [$.StrMap(a), $.Array(a)],
+      function(strMap) {
+        var ks = keys(strMap);
+        var vs = [];
+        for (var idx = 0; idx < ks.length; idx += 1) {
+          vs.push(strMap[ks[idx]]);
+        }
+        return vs;
```

Comment:
Since we're thinking about performance, I wonder whether reusing the keys array would help:

```javascript
var xs = keys(strMap);
for (var idx = 0; idx < xs.length; idx += 1) xs[idx] = strMap[xs[idx]];
return xs;
```